### PR TITLE
Integrate-s3

### PR DESCRIPTION
**Pull request by RFOxbury**
This PR brings in s3 for storing raw, interim and final outputs rather than relying on local storage. This means you can run just one script in isolation (rather than running the whole pipeline) because it can pull interim outputs from s3 :)

To test it out, run
```
python dsp_interview_transcripts/pipeline/run_pipeline.py
```
I've also used `plac` for command line arguments, so using the command above will run everything using the test paths (the alternative is to run it with the argument `--production`).

### CODE REVIEW

File: dsp_interview_transcripts/getters/data_getters.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            return float(obj)
+        elif isinstance(obj, np.integer):
+            return int(obj)
+        elif isinstance(obj, np.floating):
+            return float(obj)
+        elif isinstance(obj, np.ndarray):
+            return obj.tolist()
+        elif isinstance(obj, set):
+            return list(obj)
+        return super(CustomJsonEncoder, self).default(obj)
```
Comment (beingkk) [inline dsp_interview_transcripts/getters/data_getters.py:32]: Interesting, why did you need to add this?

File: dsp_interview_transcripts/getters/data_getters.py
Comment (beingkk) [inline dsp_interview_transcripts/getters/data_getters.py:1]: Hey, just curious if you tried first using nesta_ds_utils for the s3 functions and it wasn't fit for purpose?

File: dsp_interview_transcripts/getters/interim.py
Comment (beingkk) [inline dsp_interview_transcripts/getters/interim.py:1]: I think this could be a bit more elegant - seems a bit repetitive.

I asked ChatGPT for advice 😄  and [it suggested](https://chatgpt.com/share/e/67470b69-f268-8011-8641-b020d84bd519) creating a helper function to reduce a bit the repetition. Maybe there's something useful there!

**Review by beingkk — APPROVED** [review]
Thank you very much Rosie, it's great to have the s3 integration! 🙌 

I've added an optional suggestion about the getters.

More generally, I'm curious how do you see we could best reuse this code going forward, eg for another project? At the moment, the way the config is defined and the data stored on s3 looks specific to the present project data.

It would be a good thought exercise (for now) to consider if you have to re-use the code on a new dataset, how would you organise the work (eg, would we use configs somehow for this, or should we refactor the code into a package, and how to organise storage on s3...).

Don't need to action it, but should have an idea about this.

Thank you!!

File: dsp_interview_transcripts/getters/data_getters.py (hunk 1 above)
Comment (RFOxbury) [reply dsp_interview_transcripts/getters/data_getters.py:32]: To be honest I just copied most of these functions from [this script](https://github.com/nestauk/dap_job_quality/blob/dev/dap_job_quality/getters/data_getters.py). The CustomJsonEncoder gets used in `save_to_s3()` but it's not necessary for this project right now.

File: dsp_interview_transcripts/getters/data_getters.py
Comment (RFOxbury) [reply dsp_interview_transcripts/getters/data_getters.py:1]: No I should have tried it! I just copied functions from an existing project instead
